src/confcom/azext_confcom/README.md

```diff
@@ -143,123 +178,159 @@ Users just need to make a tar file by using the `docker save` command above, inc
+Example 13: Some use cases necessitate the use of regular expressions to allow for environment variables where either their values are secret, or unknown at policy-generation time. For these cases, the workflow below can be used:
 
 Create parameters in the ARM Template for each environment variable that has an unknown or secret value such as:
 
-    ```json
-    {
-        "parameters": {
-            "placeholderValue": {
-                "type": "string",
-                "metadata": {
-                    "description": "This value will not be placed in the template.json"
-                }
+```json
+{
+    "parameters": {
+        "placeholderValue": {
+            "type": "string",
+            "metadata": {
+                "description": "This value will not be placed in the template.json"
             }
-        },
-    }
-    ```
+        }
+    },
+}
+```
 
 Note that this parameter declaration does not have a "defaultValue". Once these parameters are defined, they may be used later in the ARM Template such as:
 
-    ```json
-    {
-        "environmentVariables": [
-            {
-                "name": "PATH",
-                "value": "/customized/path/value"
-            },
-            {
-                "name": "MY_SECRET",
-                "value": "[parameters('placeholderValue')]"
-            }
-        ]
-    }
-    ```
+```json
+{
+    "environmentVariables": [
+        {
+            "name": "PATH",
+            "value": "/customized/path/value"
+        },
+        {
+            "name": "MY_SECRET",
+            "value": "[parameters('placeholderValue')]"
+        }
+    ]
+}
+```
 
 The policy can then be generated with:
 
-    az confcom acipolicygen -a template.json
+```bash
+az confcom acipolicygen -a template.json
+```
 
 Because the ARM Template does not have a value defined for the "placeholderValue", the regular expression ".*" is used in the Rego policy. This allows for any value to be used. If the value is contained in a parameters file, that can be used when deploying such as:
 
-    az deployment group create --template-file "template.json" --parameters "parameters.json"
+```bash
+az deployment group create --template-file "template.json" --parameters "parameters.json"
+```
+
+## dmverity Layer Hashing
+
+To ensure the container that is being deployed is the intended container, the `confcom` tooling uses [dmverity hashing](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html). This is done by downloading the container locally with the Docker Daemon (or using a pre-downloaded tar file of the OCI image) and performing the dmverity hashing using the [dmverity-vhd tool](https://github.com/microsoft/hcsshim/tree/main/cmd/dmverity-vhd). These layer hashes are placed into the Rego security policy in each container's "layers" field. Note that these dmverity layer hashes are different than the layer hashes reported by `docker image inspect`.
```

Comment:
These layer hashes are placed into the Rego security policy in the "layers" field of each container.